keel-sql/src/main/kotlin/com/netflix/spinnaker/migrations/EventUidAssigner.kt

```diff
@@ -0,0 +1,73 @@
+package com.netflix.spinnaker.migrations
+
+import com.netflix.spinnaker.keel.persistence.metamodel.Tables.EVENT
+import com.netflix.spinnaker.keel.sql.inTransaction
+import de.huxhorn.sulky.ulid.ULID
+import java.time.LocalDateTime
+import java.time.ZoneOffset.UTC
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.jooq.DSLContext
+import org.jooq.Record1
+import org.slf4j.LoggerFactory
+import org.springframework.boot.context.event.ApplicationReadyEvent
+import org.springframework.context.event.EventListener
+
+class EventUidAssigner(private val jooq: DSLContext) : CoroutineScope {
+
+  private val idGenerator = ULID()
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  @EventListener(ApplicationReadyEvent::class)
+  fun onApplicationReady() {
+    launch {
+      while (countEventsWithNoUID() > 0) {
+        var count = 0
+        jooq.inTransaction {
+          select(EVENT.TIMESTAMP)
+            .from(EVENT)
+            .where(EVENT.UID.isNull)
+            .limit(25)
```

Comment:
I did a bit of back of a napkin perf analysis and this batch size seemed optimal for not holding the transaction open too long. It can do 25 in about a second on my machine. Bumping to 100 makes it take ~10 seconds.

Reply:
Actually, it's running a ton faster locally now so who knows if this could be further optimized? It's not really important though.

Reply:
Honestly, smaller batches for this migration process is probably better than more anyway.